internal/v2/cert_verifier/cert_verifier.go

```diff
@@ -65,6 +65,11 @@ func VerifyClientCertificateChain(verificationMode s2av2pb.ValidatePeerCertifica
 			return fmt.Errorf("failed to offload client cert verification to S2A: %d, %v", resp.GetStatus().Code, resp.GetStatus().Details)
 
 		}
+
+		if resp.GetValidatePeerCertificateChainResp().ValidationResult != s2av2pb.ValidatePeerCertificateChainResp_SUCCESS {
```

Comment:
Please add a unit test to hit this case.

Reply:
Will do!
The way I'll do this is as follows:
- To get coverage on this line, I'll modify the fake s2av2 service layer to populate the SessionResp and ValidationResult/ValidationDetails separately
- add a unit test that will test the SessionResp Status field
- add a unit test that will test the ValidationResult field